contribs/drt-extensions/src/main/java/org/matsim/contrib/drt/extension/operations/eshifts/charging/ChargingBreakActivity.java

```diff
@@ -95,9 +98,9 @@ protected void simStep(double now) {
         chargingDelegate.doSimStep(now);
     }
 
-    private AcceptedDrtRequest getRequestForPassenger(Id<Person> passengerId) {
+    private AcceptedDrtRequest getRequestForPassengers(Set<Id<Person>> passengerIds) {
         return pickupRequests.values().stream()
-                .filter(r -> passengerId.equals(r.getPassengerId()))
+                .filter(r -> r.getPassengerIds().containsAll(passengerIds))
```

Comment:
This will ensure the two lists contain exactly the same elements (assuming there are no duplicates) and it should be quicker to discard the requests if many of them have different sizes.
```java
r.getPassengerIds() == passengerIds.size() && r.getPassengerIds().containsAll(passengerIds)
```